zerver/lib/actions.py

```diff
@@ -4529,9 +4532,14 @@ def do_remove_realm_custom_profile_field(realm: Realm, field: CustomProfileField
     notify_realm_custom_profile_fields(realm, 'delete')
 
 def try_update_realm_custom_profile_field(realm: Realm, field: CustomProfileField,
-                                          name: Text) -> None:
+                                          name: Text,
+                                          hint: Optional[Text]=None) -> None:
+    to_update = ['name']
     field.name = name
-    field.save(update_fields=['name'])
+    if hint is not None:
+        field.hint = hint
+        to_update.append('hint')
+    field.save(update_fields=to_update)
     notify_realm_custom_profile_fields(realm, 'update')
```

Comment:
I think it might be cleaner to just have 2 functions, one for name and one for hint.